arangod/Replication2/StateMachines/Document/Actor/Scheduler.cpp

```diff
@@ -0,0 +1,46 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Manuel Pöter
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Scheduler.h"
+
+#include "Assertions/Assert.h"
+#include "Scheduler/Scheduler.h"
+
+namespace arangodb::replication2::replicated_state::document::actor {
+
+Scheduler::Scheduler(std::shared_ptr<replication2::IScheduler> scheduler)
+    : _scheduler{std::move(scheduler)} {
+  TRI_ASSERT(_scheduler != nullptr);
+}
+
+void Scheduler::queue(arangodb::actor::LazyWorker&& worker) {
+  _scheduler->queue(std::move(worker));
+}
+
+void Scheduler::delay(std::chrono::seconds delay,
+                      std::function<void(bool)>&& fn) {
+  std::ignore =
```

Comment:
Yes, I found this recently, fortunately this is not an issue for us at the moment, because we are not using delay in the replication2 code. However, we cannot remove this method because it is used in the new pregel implementation, and fixing it is unfortunately also not trivial.

Reply:
Okay, then i would suggest we add a `TRI_ASSERT(false) << "Called not implemented method"` here. This way we are sure no one uses this, also not in the Future

Reply:
Ah forget it, this is the shared method.

Reply:
Yes, but the actor based pregel implementation will probably be removed soon, then we could also completely remove this method. For now I would suggest to ignore this, since it is not relevant for this PR.